prioritization_fees/src/account_priofees.rs

```diff
@@ -178,24 +190,41 @@ impl AccountPrioStore {
         let min_slot_to_retain = produced_block
             .slot
             .saturating_sub(self.number_of_slots_to_save as u64);
-        let cleanup_functor = |mut iter: RefMutMulti<'_, Pubkey, AccountPrio>| {
+        let cleanup_functor = |iter: &mut AccountPrio| {
             while let Some((k, _)) = iter.stats_by_slot.first_key_value() {
                 if *k > min_slot_to_retain {
                     break;
                 }
                 iter.stats_by_slot.pop_first();
             }
         };
-        self.account_by_prio_fees_all
-            .iter_mut()
-            .for_each(cleanup_functor);
+        self.account_by_prio_fees_all.retain(|_, ap| {
+            cleanup_functor(ap);
+            !ap.stats_by_slot.is_empty()
+        });
+
+        self.account_by_prio_fees_writeonly.retain(|_, ap| {
+            cleanup_functor(ap);
+            !ap.stats_by_slot.is_empty()
+        });
+
+        ACCOUNTS_IN_PRIO_CACHE.set(self.account_by_prio_fees_all.len() as i64);
+        WRITE_ONLY_ACCOUNTS_IN_PRIO_CACHE.set(self.account_by_prio_fees_writeonly.len() as i64);
+        let dp_all: usize = self
+            .account_by_prio_fees_all
+            .iter()
+            .map(|v| v.stats_by_slot.len())
+            .sum();
+        let dp_writeonly: usize = self
+            .account_by_prio_fees_writeonly
+            .iter()
+            .map(|v| v.stats_by_slot.len())
+            .sum();
+        NUMBER_OF_PRIO_DATA_POINTS.set((dp_all + dp_writeonly) as i64);
```

Comment:
is that correct? _all_ should include _writeonly_ - no?

Reply:
No we mangage for both all and writeonly